lib/util/exec/hermit.ts

```diff
@@ -0,0 +1,90 @@
+import os from 'os';
+import fs from 'fs-extra';
+import upath from 'upath';
+import { GlobalConfig } from '../../config/global';
+import { logger } from '../../logger';
+import { rawExec } from './common';
+import type { RawExecOptions } from './types';
+
+export function isHermit(): boolean {
+  const { binarySource } = GlobalConfig.get();
+  return binarySource === 'hermit';
+}
+
+async function statFile(f: string): Promise<fs.Stats | undefined> {
+  let exists: fs.Stats | undefined = undefined;
+  try {
+    exists = await fs.stat(f);
+  } catch (e) {
+    // not doing anything when file not exists for errors
+  }
+
+  return exists;
+}
+
+async function statHermit(
+  defaultCwd: string,
+  parts: string[]
+): Promise<fs.Stats | undefined> {
+  const hermitForCwd = upath.join(...[defaultCwd, ...parts, 'bin', 'hermit']);
+  logger.trace({ hermitForCwd }, 'looking up hermit');
+  return await statFile(hermitForCwd);
+}
+
+export async function findHermitCwd(cwd: string): Promise<string> {
+  const defaultCwd = GlobalConfig.get('localDir') ?? '';
+  const parts = cwd.replace(defaultCwd, '').split(upath.sep);
+  let exists: fs.Stats | undefined = undefined;
+
+  // search the current relative path until reach the defaultCwd
+  while (parts.length > 0) {
+    exists = await statHermit(defaultCwd, parts);
+    // on file found. break out of the loop
+    if (exists !== undefined) {
+      break;
+    }
+    // otherwise, continue searching in parent directory
+    parts.pop();
+  }
+
+  // search in defaultCwd
+  if (exists === undefined) {
+    exists = await statHermit(defaultCwd, parts);
+  }
+
+  if (exists === undefined) {
+    throw new Error(`hermit not found for ${cwd}`);
```

Comment:
if hermit not found for cwd, does that mean that we should get out of here completely?
where does this error get caught in a real repository? do you think you have to catch the error somewhere?
this might throw off the whole job run

Reply:
Getting out of here is reasonable from hermit's point of view.

If we don't throw the error here, down the road it will failed the `hermit env` command. Without `hermit env` being correctly plugged the execution might fail unless the binary is available in the running env, which kinda breaks the purpose of using hermit as a binary source. 

I'm not sure where the error get caught in a real repo. But It is similar to the situation for `binarySource=docker`  where `docker pull` failed.  

Not knowing how it would be handled in other places, but for `updateArtifacts` specifically the error is caught in the `processBranch` method which doesn't look like it is only cancels the job for the branch in a specific repository?